unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -613,5 +618,28 @@ export function CategoryFacetTest() {
         expect(ellipsis.nextSibling.textContent).toContain('parent25');
       });
     });
+
+    describe('testing the DependsOnManager', () => {
+      beforeEach(() => {
+        spyOn(test.cmp.dependsOnManager, 'updateVisibilityBasedOnDependsOn');
+        spyOn(test.cmp.dependsOnManager, 'listenToParentIfDependentFacet');
+      });
+
+      it('should initialize the dependsOnManager', () => {
+        expect(test.cmp.dependsOnManager).toBeTruthy();
+      });
+
+      it('should call the "listenToParentIfDependentFacet" method of the DependsOnManager on initialization', () => {
+        spyOn(DependsOnManager.prototype, 'listenToParentIfDependentFacet');
```

Comment:
🤔 What's the reason for spying on the prototype function instead of on an instance method in all your tests ?

Reply:
Do you know how to spy on a method called in the `constructor` of a class, I can only `spyOn` after it's been initialized it seems... That's why spying on the prototype did the trick.